detox/src/realms/DetoxContext.js

```diff
@@ -0,0 +1,180 @@
+const funpermaproxy = require('funpermaproxy');
+
+const temporary = require('../artifacts/utils/temporaryPath');
+const { DetoxRuntimeError } = require('../errors');
+const { DetoxLogger, installLegacyTracerInterface } = require('../logger');
+const symbols = require('../symbols');
+
+const DetoxConstants = require('./DetoxConstants');
+
+const $cleanup = Symbol('cleanup');
+const $restoreSessionState = Symbol('restoreSessionState');
+const $sessionState = Symbol('restoreSessionState');
+const $status = Symbol('status');
+const $worker = Symbol('worker');
```

Comment:
We use symbols to prevent external users from accessing private and protected methods.

E.g. you can't call `require('detox').init()`, but you can call `require('detox/internals').init()`, although technically both the files refer to the same Detox context instance.

Reply:
I wish IDEs were a bit smarter (I also use WebStorm, btw), but I prefer to suffer here than to allow any possibility of fiddling with the most public objects Detox exports.